cmd/root_test.go

```diff
@@ -27,6 +29,19 @@ type globalTestState struct {
 	expectedExitCode int
 }
 
+func getFreePort(t *testing.T) int {
+	freePortAddr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	require.NoError(t, err)
+	l, err := net.ListenTCP("tcp", freePortAddr)
```

Comment:
This is a nitpick on something that's very unlikely to happen:  to be 100% safe, I think it might be better to let `ListenTCP` do the automatic port binding and then retrieve the port from its `TCPAddr.` 
As opposed to the first resolve and then binding. That is in the unlikely (?) event that another test using this code has already bounded the selected port in the meantime.

Example of what I mean [in xk6-redis](https://github.com/grafana/xk6-redis/blob/master/redis/stub_test.go#L67)

Reply:
Yeah, that could be a problem, though I did something a bit different in the [latest force-push](https://github.com/grafana/k6/compare/a6823b208ee3353e499354e2fc5f2fe6dcb61dc8..b37b0dd69a8f5e11806e202e001f2d490452e5a9) to fix it, which should be even more resilient :crossed_fingers: 

The port racing problem you mentioned was real, but it wouldn't have been solved even by your suggestion, since we are only configuring the free address in the k6 config here, there might be several hundred milliseconds between that action and the `k6 run` actually binding to that port. So, some other test might get the same port number in the meantime.

So I decided to sequentially use a global variable and an atomic operation in the tests, to sequentially look for open ports and give them out to any test that requests them. This way, there shouldn't be any chance of two integration tests trying to use the same port. Some other test might get in the way, though there is not a lot we can do about that, unfortunately :disappointed:

Reply:
Very good 👍🏻